components/rollup_node/block_gen.md

```diff
@@ -0,0 +1,147 @@
+# Block Generation
+
+The logic which is used to generate the rollup chain from L1.
+
+## Glossary
+
+- **Block Inputs**: All information required to generate a full Ethereum block, given the pre-state and previous block. For example, does NOT include a state root for that block's input.
+- **Rollup Blockchain**: The rollup blocks generated by performing the rollup's state transition function over all block inputs. For example, DOES include a state root for that block's input.
+
+## Summary
+
+The rollup chain can be deterministically generated given an L1 Ethereum chain. The fact that the entire rollup chain can be generated based on L1 blocks is _what makes OE a rollup_. This process can be represented as:
+
+```python
+f(l1_blockchain) -> rollup_blockchain
+```
+
+In this document we define a block generation function which is designed to be:
+
+1. Context independent given a constant sized subset of the L1 chain.
```

Comment:
> Is this intended to mean that the generation function is a pure function of the L1 chain subset?

Yes that's right. However, I will modify this definition to instead be something along the lines of:

1. Require no state other than what is easily accessible in an L1 EE or L2 EE.

The key property that I really wanted was that the rollup driver doesn't impose **additional** state on top of L1 / L2. This way we don't need to "sync" the rollup driver and maintain a special database for it.

Reply:
> Require no state other than what is easily accessible in an L1 EE or L2 EE.

What kind of "or" is it:
1. We can require state available on L1, or state available on L2 (but not (yet?) on L1), or
2. We can require state that is available on L1 and on L2?

Reply:
Calculating the block for epoch `N` should be possible with only:
1) The data from the recent L1 blocks `[N - sequencing_window, N]`.
2) The data or state from L2 after epoch `N-1`.

In other words, "easily accessible" basically means readable.  So if there are multiple L2 blocks inputs for epoch, you should not have to execute any of them to know all of them.

>  state available on L2 (but not (yet?) on L1)

If by this you mean does the data need to be finalized (passed challenge window) on L1? If so, the answer is no.  This data is accessible in the fraud proof will build on a previous assertion, meaning that it can accept the L2 state from the last epoch as an input.

Reply:
I meant state not yet posted on L1, but it seems to me this is purely a L1 thing, I was just tripped up by the reformulation to "Require no state other than what is easily accessible in an L1 EE or L2 EE."